src/components/CharacterGallery.js

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import { withRouter } from 'react-router-dom';
+import { getCharacters } from '../util/swapi';
+import CharacterItem from './CharacterItem';
+import './CharacterGallery.css';
+
+class CharacterGallery extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            characters: [],
+            selectedGender: 'all'
+        };
+    }
+
+    async componentDidMount() {
+        const characters = await getCharacters(this.props.urls);
+        this.setState({ characters: characters });
+    }
+
+    showMaleGender = () => {
+        this.setState({ selectedGender: 'male' });
+    }
+
+    showFemaleGender = () => {
+        this.setState({ selectedGender: 'female' });
+    }
+
+    showAllGenders = () => {
+        this.setState({ selectedGender: 'all' });
+    }
+
+    makeCharacterItems() {
+        const characterItems = [];
+
+        for (let i = 0; i < this.state.characters.length; i++) {
+            characterItems.push(
+                <CharacterItem
+                    key={this.state.characters[i].name}
+                    selectedGender={this.state.selectedGender}
+                    character={this.state.characters[i]}
```

Comment:
Not to have long variables we should destructrure this.state and take the relevant values: characters and selectedGender.